packages/codefixes/src/fixes/glint/glint-parsing-utils.ts

```diff
@@ -243,3 +243,57 @@ export function getJSDocExtendsTagWithSignature(
 
   return;
 }
+
+type LikeTemplateOnlyComponentVariableDeclaration = ts.VariableDeclaration & {
+  type: ts.TypeReferenceNode & { typeName: ts.Identifier };
+};
+
+function isTemplateOnlyComponent(
+  targetNode: ts.Node,
+  identifierName = 'TemplateOnlyComponent'
+): targetNode is LikeTemplateOnlyComponentVariableDeclaration {
+  if (
+    ts.isVariableDeclaration(targetNode) &&
+    targetNode.type &&
+    ts.isTypeReferenceNode(targetNode.type) &&
+    ts.isIdentifier(targetNode.type.typeName) &&
+    targetNode.type.typeName.escapedText === identifierName
+  ) {
+    return true;
+  }
+
+  return false;
+}
+
+export function getNearestTemplateOnlyComponentVariableDeclaration(
+  targetNode: ts.Node,
+  identifierName = 'TemplateOnlyComponent'
+): LikeTemplateOnlyComponentVariableDeclaration | undefined {
+  let target = targetNode;
+  do {
+    if (!target.parent) {
+      return;
+    }
+
+    if (isTemplateOnlyComponent(target, identifierName)) {
+      return target;
+    }
+  } while ((target = target.parent));
```

Comment:
You might want to try ts.findAncestor, like this: https://github.com/rehearsal-js/rehearsal-js/blob/master/packages/codefixes/src/fixes/addMissingTypesBasedOnInheritance.ts#L65

Reply:
Thank you for the recommendation. Will update this to use the library method.

Reply:
I've pushed up  changes to the two methods that were finding up.